crawler.js

```diff
@@ -114,7 +114,13 @@ class Crawler {
     }
 
     this.configureUA();
-
+    
+    // Check that the collection name is valid.
+    if (this.params.collection.search(/^[a-zA-Z0-9_-]*$/) === -1){
```

Comment:
Let's move to this to `validateArgs()` function and also throw an error, similar to other param validation